src/server/pfs/server/api_server.go

```diff
@@ -355,6 +355,14 @@ func (a *apiServer) putFileObj(objClient obj.Client, request *pfs.PutFileRequest
 				defer func() {
 					<-sem
 				}()
+
+				if strings.HasSuffix(name, "/") {
+					// Amazon S3 supports objs w keys that end in a '/'
+					// PFS needs to treat such a key as a directory.
+					// In this case, we rely on the driver PutFile to
+					// construct the 'directory' diffs from the file prefix
+					return nil
```

Comment:
We should probably have some sort of log message about the fact that we're omitting this file.